command/snapshot/inspect/snapshot_inspect.go

```diff
@@ -28,16 +24,41 @@ func New(ui cli.Ui) *cmd {
 }
 
 type cmd struct {
-	UI    cli.Ui
-	flags *flag.FlagSet
-	help  string
+	UI     cli.Ui
+	flags  *flag.FlagSet
+	help   string
+	format string
 }
 
 func (c *cmd) init() {
 	c.flags = flag.NewFlagSet("", flag.ContinueOnError)
+	c.flags.StringVar(
+		&c.format,
+		"format",
+		PrettyFormat,
+		fmt.Sprintf("Output format {%s}", strings.Join(GetSupportedFormats(), "|")))
+
 	c.help = flags.Usage(help, c.flags)
 }
 
+// MetadataInfo is used for passing information
+// through the formatter
+type MetadataInfo struct {
+	ID      string
+	Size    int64
+	Index   uint64
+	Term    uint64
+	Version raft.SnapshotVersion
+}
+
+// OutputFormat is used for passing information
+// through the formatter
+type OutputFormat struct {
+	Meta      *MetadataInfo
+	Stats     map[structs.MessageType]typeStats
```

Comment:
I wonder whether this field should be `Stats []typeStats`. In the pretty output I see that you are creating a slice of these and then sorting. For the JSON output I wonder whether it would be better to see:

```
"Stats": [
   {
      "Name": "Register",
      "Count": 20,
      "Sum": 1537,
   }
]
```

Having the MessageType (`int`) as a map key in JSON doesn't seem that useful. Also having it in an array would more easily allow us to aggregate stats for namespaces without having the map keys conflict.

Reply:
Oh I didn't even notice that, good catch.